riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/TestRiptideRegistrar.java

```diff
@@ -0,0 +1,53 @@
+package org.zalando.riptide.spring;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.http.client.AsyncClientHttpRequestFactory;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.AsyncRestTemplate;
+
+import static org.springframework.beans.factory.support.BeanDefinitionBuilder.genericBeanDefinition;
+
+@Slf4j
+@AllArgsConstructor
+class TestRiptideRegistrar implements RiptideRegistrar {
+
+    private final Registry registry;
+    private final RiptideSettings settings;
+
+    @Override
+    public void register() {
+        final String templateId = registerMockAsyncRestTemplate();
+        final String serverId = registerMockRestServiceServer(templateId);
+        settings.getClients().forEach((id, client) -> registerAsyncClientHttpRequestFactory(id, templateId, serverId));
+    }
+
+    private String registerMockAsyncRestTemplate() {
```

Comment:
Couldn't you register the template and the server using traditional `@Bean` methods in the `RiptideTestAutoConfiguration`?

Reply:
Technically yes, but I haven't found a cool way to still reference these beans in here on registering the request factories. Just by some explicit bean name? Any idea? I feel those `2 + (1 * n)` beans should go together.

Reply:
Explicit bean name should work. I'd argue that the configuration binds those 2+n beans together. The Registrar is just our way to dynamically register the n beans.